src/responder/kcm/kcmsrv_cmd.c

```diff
@@ -240,7 +239,6 @@ struct kcm_repbuf {
     uint8_t rcbuf[KCM_RETCODE_SIZE];
     struct kcm_iovec v_rc;
 
-    uint8_t msgbuf[KCM_PACKET_MAX_SIZE];
```

Comment:
This is not a hard nack, but more of a suggestion or a question, but one reason I introduced the kcm_reqbuf/kcm_repuf structures was to have a single structure that would hold everything. Wouldn't it be cleaner to keep the pointer to the allocated message in these structures and owned, talloc-wise by the structure?

Reply:
I've decided to remove it because keeping it updated is just a work that doesn't seem logical enough when we're dealing with malloc instead of using a pre-allocated vector.

For instance ... on kcm_recv_data() we'd have to do (considering we'd also add a msglen field to the struct):
```
reqbuf->msgout = reqbuf->v_msg.kiov_base = msg;
 reqbuf->msglen =  reqbuf->v_msg.kiov_len = msglen;
```

And then where else it would be used?

When reading the code I understood your idea and I do think it made things simpler when dealing with the memory that's in the heap. But I couldn't see any benefit now we're dealing with the stack.

Another option would be to do a first talloc when initializing the structures and then realloc the bufmsg accordingly when we know the needed size, but it also seems one more complication for no real benefit.

Now the question ... shall I get back to use msgbuf?

Reply:
So, I don''t need the pointer in the reqbuf/repbuf structure, but what I'm wary about is a clean talloc hierarchy. In that respect, what do you think about allocating the rep pointer atop repbuf in kcm_output_construct or allocating it the way you do now and stealing it atop repbuf as you assign the pointer?

Reply:
re[pq]buf are not talloc contexts, we can't go for your suggestion.
What we could do is pass down the kcm_req_context and use it as the context instead of cctx.

Would you prefer using kcm_req_ctx instead of cctx as memory context for the payloads?